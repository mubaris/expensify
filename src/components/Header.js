import React from 'react';
import { NavLink } from 'react-router-dom';
import convertToEmoji from './../emoji';

const Header = () => {
  const emojifiedTitle = convertToEmoji('Expensify 💰');
  const helpText = convertToEmoji('Help 💁');
  return (
    <header>
      <h1 dangerouslySetInnerHTML={{ __html: emojifiedTitle }} />
      <NavLink to="/" activeClassName="is-active" exact>
                Dashboard
      </NavLink>
      <NavLink to="/create" activeClassName="is-active">
                Create
      </NavLink>
      <NavLink
        activeClassName="is-active"
        to="/help"
        dangerouslySetInnerHTML={{ __html: helpText }}
      />
    </header>
  );
};

export default Header;
