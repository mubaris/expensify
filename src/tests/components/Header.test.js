import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

test('Should render Header correctly', () => {
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
