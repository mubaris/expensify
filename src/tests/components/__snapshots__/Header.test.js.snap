// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1
    dangerouslySetInnerHTML={
      Object {
        "__html": "Expensify <img src=\\"/emoji-data/img-apple-64/1f4b0.png\\" class=\\"emoji\\" data-codepoints=\\"1f4b0\\" />",
      }
    }
  />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Help <img src=\\"/emoji-data/img-apple-64/1f481-200d-2640-fe0f.png\\" class=\\"emoji\\" data-codepoints=\\"1f481-200d-2640-fe0f\\" />",
      }
    }
    to="/help"
  />
</header>
`;
